src/lib/tls/tls_callbacks.h

```diff
@@ -45,50 +45,98 @@ class BOTAN_PUBLIC_API(2, 0) Callbacks {
    public:
       virtual ~Callbacks() = default;
 
-      /**
-       * Mandatory callback: output function
-       * The channel will call this with data which needs to be sent to the peer
-       * (eg, over a socket or some other form of IPC). The array will be overwritten
-       * when the function returns so a copy must be made if the data cannot be
-       * sent immediately.
+       /**
+       * @name Mandatory
        *
-       * @param data a contiguous data buffer to send
+       * Those callbacks must be implemented by all applications that use TLS.
+       * @{
        */
+
+      /**
+      * Mandatory callback: output function
+      *
+      * The channel will call this with data which needs to be sent to the peer
+      * (eg, over a socket or some other form of IPC). The array will be overwritten
+      * when the function returns so a copy must be made if the data cannot be
+      * sent immediately.
+      *
+      * As an example you could use the syscall ``send`` to perform a blocking
+      * write on a socket, or append the data to a queue managed by your
+      * application and initiate an asynchronous write.
+      *
+      * For TLS all writes must occur *in the order requested*. For DTLS this
+      * ordering is not strictly required, but is still recommended.
+      *
+      * @param data a contiguous data buffer to send
+      */
       virtual void tls_emit_data(std::span<const uint8_t> data) = 0;
 
       /**
-       * Mandatory callback: process application data
-       * Called when application data record is received from the peer.
-       * Again the array is overwritten immediately after the function returns.
-       *
-       * @param seq_no the underlying TLS/DTLS record sequence number
-       *
-       * @param data a contiguous data buffer containing the received record
-       */
+      * Mandatory callback: process application data
+      *
+      * Called when application data record is received from the peer. Again
+      * the array is overwritten immediately after the function returns.
```

Comment:
```suggestion
      * Called when application data record is received from the peer. The
      * array is overwritten immediately after the function returns.
```

... can't really assume that people read this cover-to-cover.